### PR TITLE
Basic template to deploy an Azure Databricks workspace

### PR DESCRIPTION
**Pull request by jeffpang**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Initial submission

### CODE REVIEW

File: 101-databricks-workspace/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      },
+      "properties": {
+        "ManagedResourceGroupId": "[concat(subscription().id, '/resourceGroups/', variables('managedResourceGroupName'))]"
+      }
+    }
+  ],
+  "outputs": {
+    "workspace": {
+      "type": "object",
+      "value": "[reference(resourceId('Microsoft.Databricks/workspaces',parameters('workspaceName')))]"
```
Comment (yunpark93) [inline 101-databricks-workspace/azuredeploy.json]: Nit: space after comma

File: 101-databricks-workspace/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "workspaceName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Azure Databricks workspace to create."
+      }
+    },
+    "workspaceTier": {
```
Comment (yunpark93) [inline 101-databricks-workspace/azuredeploy.json]: Call this pricingTier to make it consistent with Portal Blade

**Comment by mgyucht** [comment]
LGTM!

File: 101-databricks-workspace/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    {
+      "type": "Microsoft.Databricks/workspaces",
+      "name": "[parameters('workspaceName')]",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2018-04-01",
+      "sku": {
+        "name": "[parameters('pricingTier')]"
+      },
+      "properties": {
+        "ManagedResourceGroupId": "[concat(subscription().id, '/resourceGroups/', variables('managedResourceGroupName'))]"
```
Comment (bmoore-msft) [inline 101-databricks-workspace/azuredeploy.json]: is this meant to point to a resourceGroup that doesn't exist?

File: 101-databricks-workspace/azuredeploy.parameters.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+          "value": "GEN-UNIQUE"
+        },
+        "pricingTier": {
```
Comment (bmoore-msft) [inline 101-databricks-workspace/azuredeploy.parameters.json]: remove this and just use the defaultValue in the template

File: 101-databricks-workspace/azuredeploy.json (hunk 3 above)
Comment (jeffpang) [reply 101-databricks-workspace/azuredeploy.json]: Yep. Databricks is implemented using Azure Managed Applications so it creates a new resourceGroup as part of its deployment.

File: 101-databricks-workspace/azuredeploy.parameters.json (hunk 4 above)
Comment (jeffpang) [reply 101-databricks-workspace/azuredeploy.parameters.json]: Done.

**Comment by jeffpang** [comment]
@bmoore-msft I think I've addressed your comments. Let me know if anything else is needed.

**Pull request merged**